[docs] Modules overview updates (#6774)
deckhouse/deckhouse: docs/site/backends/docs-builder-template/content/modules/_index.md

Deckhouse Platform has a modular structure. Modules can be embedded in Deckhouse or connected using the `ModuleSource` resource.

> The main difference of the _embedded_ Deckhouse module is that the embedded module is delivered as part of the Deckhouse platform and shares a common release cycle with Deckhouse. Documentation for Deckhouse embedded modules can be found in the [Deckhouse documentation](/documentation/v1/) section.

Reviewer: ```suggestion
The distinguishing feature of an _embedded_ Deckhouse module is that it is delivered as part of the Deckhouse platform and shares a common Deckhouse release cycle. For more information on embedded Deckhouse modules, see the [Deckhouse documentation](/documentation/v1/).
```

> Deckhouse modules (connected using the `ModuleSource` resource) have a release cycle that is independent of Deckhouse, i. e. they can be updated independently of Deckhouse versions. Deckhouse modules may be developed by a team that is not part of the Deckhouse development team. The operation of a particular module may affect the stability of Deckhouse, but we strive to ensure that such an impact will not have serious consequences for the platform as a whole. 

Author: ```suggestion
Deckhouse modules (connected using the `ModuleSource` resource) have a release cycle that is independent of Deckhouse, i. e. they can be updated independently of Deckhouse versions. Deckhouse modules may be developed by a team that is not part of the Deckhouse development team. The operation of a particular module may affect the stability of Deckhouse, but we strive to ensure that such an impact will not have serious consequences for the platform as a whole.
```

> Deckhouse modules (connected using the `ModuleSource` resource) have a release cycle independent of Deckhouse, i.e. they can be updated independently of Deckhouse versions. Deckhouse modules can be developed by a development team separate from the Deckhouse development team. The operation of a particular module may impact Deckhouse stability, although we strive to ensure that this impact does not have serious consequences for the entire platform.

Reviewer: ```suggestion
Deckhouse modules (connected using the `ModuleSource` resource) have a release cycle that is independent of Deckhouse, i. e. they can be updated independently of Deckhouse versions. Deckhouse modules may be developed by a team that is not part of the Deckhouse development team. The operation of a particular module may affect the stability of Deckhouse, but we strive to ensure that such an impact will not have serious consequences for the platform as a whole. 
```

> This section provides information on Deckhouse modules that have passed preliminary compatibility testing and are approved for use in conjunction with Deckhouse.

Reviewer: ```suggestion
This section provides information on Deckhouse modules that have passed preliminary compatibility testing and have been approved for use with Deckhouse.
```
